Registration form validations

kuosmark opened the pull request:
# Description

Fixes:
- [Validate username from server and show error notification](https://trello.com/c/GCRXqKrv/671-registration-page-validate-username-from-server-and-show-error-notification)
- [Length validations for registration form fields](https://trello.com/c/PnlcOYgq/700-length-validations-for-registration-form-fields)
- [Password field validations](https://trello.com/c/tbGGIIHW/702-password-field-validations)
- [Generic Network Error](https://trello.com/c/oEudB7tz/706-generic-network-error)

## Why was the change made?

We want to tell the user to edit invalid fields sending the form. We also want to explain why. This improves user experience.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unittest
- [x] Cypress e2e -tests

# Caveats?

Does this introduce new warnings or are linter rules suppressed? Why? Is only manual testing applicable for this feature? Is there something special that the reviewer should take into account?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

rottabonus reviewed (commented):
Code looks good! =) 

Few comments:

1. When running e2e tests, the _shows error message if username is taken_ fails against an empty db, because the takenUsername-user is not in db. Consider adding a helper function to create that user before the test runs
2. Consider cleaning some of the history, because there are quite a lot of commits. Maybe some redundant "Fix typo things" could be squashed. This is a nicpicky thing really
3. Maybe update the command to run cypress test on the README (API runs on 8080, we run the cypress tests on 8081)
4. More of a question.. Do you think localizations for the registration-page will be a PITA? This is something we need to start to think about already =P

I wont approve because the e2e tests fail, that is a hard requirement. :p

kuosmark commented:
1. The user with takenUsername is now created in the test suite. I also updated the Readme instructions for running the tests locally (clearing the db before running the suite again).
2. I'll take this into account in the future, now I just want this PR over with. :D
3. Updated!
4. I don't think it will be that bad. We'll just have all the different language text options in the html file and have a function that switches which ones we show when pressing some "change language" button.

rottabonus reviewed (approved):
Nice!
